state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -2249,44 +2282,53 @@ pub mod pallet {
 		}
 	}
 
-	impl<T: Config> cf_traits::AssetConverter for Pallet<T> {
+	impl<T: Config> AssetConverter for Pallet<T> {
 		fn calculate_input_for_gas_output<C: Chain>(
 			input_asset: C::ChainAsset,
 			required_gas: C::ChainAmount,
 		) -> Option<C::ChainAmount> {
+			Self::calculate_input_for_desired_output(
+				input_asset.into(),
+				C::GAS_ASSET.into(),
+				required_gas.into(),
+				true,
+			)
+			.map(|amount| C::ChainAmount::try_from(amount).expect("Asset amount is for this chain"))
```

Comment:
Is it really true this can't panic? The asset amount is specifically not for this chain before this point right? It's a general asset amount that could be outside the bounds of a particular ChainAmount when we get to here. For example at a time where the liquidity is way off, it could mean we require a huge amount of input asset to get the output asset, which might fit AssetAmount, but not ChainAmount.

Reply:
Would be nice to have a test for this case.